src/data/util.py

```diff
@@ -25,61 +25,106 @@ def batched(iterable, n):
+    """An enum representing the results of a game.
     Should be subclassed if the point value is not the default"""
     NOT_PAIRED = 0
     LOSS = 1
-    DRAW_OR_HPB = 2  # HPB = Halp Point Bye
+    DRAW = 2
     GAIN = 3
     FORFEIT_LOSS = 4
     DOUBLE_FORFEIT = 5
-    PAB_OR_FORFEIT_GAIN_OR_FPB = 6  # PAB = Pairing-Allocated-Bye, FPB = Full Point Bye
+    FORFEIT_GAIN = 6
+    HALF_POINT_BYE = 7
+    PAIRING_ALLOCATED_BYE = 8
+    FULL_POINT_BYE = 9
+    UNRATED_LOSS = 10
+    UNRATED_DRAW = 11
+    UNRATED_GAIN = 12
 
     def __str__(self) -> str:
         match self:
-            case Result.GAIN:
+            case Result.GAIN | Result.UNRATED_GAIN:
                 return '1-0'
-            case Result.LOSS:
+            case Result.LOSS | Result.UNRATED_LOSS:
                 return '0-1'
-            case Result.DRAW_OR_HPB:
+            case Result.DRAW | Result.UNRATED_DRAW | Result.HALF_POINT_BYE:
                 return '1/2'
             case Result.NOT_PAIRED:
                 return ''
             case Result.FORFEIT_LOSS:
                 return 'F-1'
-            case Result.PAB_OR_FORFEIT_GAIN_OR_FPB:
+            case Result.FORFEIT_GAIN | Result.PAIRING_ALLOCATED_BYE | Result.FULL_POINT_BYE:
                 return '1-F'
             case Result.DOUBLE_FORFEIT:
                 return 'F-F'
             case _:
                 raise ValueError(f'Unknown value: {self}')
 
     @classmethod
-    def from_papi_value(cls, value: int) -> Self:
+    def from_papi_value(
+            cls,
+            value: int,
+            is_point_bye: bool = False,
+            is_pairing_bye: bool = False,
+            is_unrated: bool = False) -> Self:
         """Create a `Result` instance from the stored value in the
         Papi database."""
         match value:
-            case 0:
+            case PapiResult.NOT_PAIRED.value:
                 return cls.NOT_PAIRED
-            case 1:
-                return cls.LOSS
-            case 2:
-                return cls.DRAW_OR_HPB
-            case 3:
-                return cls.GAIN
-            case 6:
-                return cls.PAB_OR_FORFEIT_GAIN_OR_FPB
-            case 4:
+            case PapiResult.LOSS.value:
+                return cls.UNRATED_LOSS if is_unrated else cls.LOSS
+            case PapiResult.DRAW_OR_HPB.value:
+                return cls.UNRATED_DRAW if is_unrated else cls.HALF_POINT_BYE if is_point_bye else cls.DRAW
+            case PapiResult.GAIN.value:
+                return cls.UNRATED_GAIN if is_unrated else cls.GAIN
+            case PapiResult.PAB_OR_FORFEIT_GAIN_OR_FPB.value:
+                return (
+                    cls.FULL_POINT_BYE if is_point_bye
+                    else cls.PAIRING_ALLOCATED_BYE if is_pairing_bye
+                    else cls.FORFEIT_GAIN)
+            case PapiResult.FORFEIT_LOSS:
                 return cls.FORFEIT_LOSS
-            case 5:
+            case PapiResult.DOUBLE_FORFEIT:
```

Comment:
Shouldn't those be `PapiResult.FORFEIT_LOSS.value` and `PapiResult.DOUBLE_FORFEIT.value` instead?